backend/tests/integration/submissions/test_submissions_views.py

```diff
@@ -1446,39 +1479,49 @@ def test_new_seminar_submission_with_cataloger_login(
     send_confirmation_mock.assert_not_called()
 
 
+@pytest.mark.parametrize(
+    "form_data,record_data",
+    [
+        (SEMINAR_FORM_DATA, SEMINAR_RECORD_DATA),
+        (REQUIRED_SEMINAR_FORM_DATA, REQUIRED_SEMINAR_RECORD_DATA),
+    ],
+)
 @patch("inspirehep.submissions.views.async_create_ticket_with_template")
-def test_seminar_update_submission(create_ticket_mock, inspire_app):
+def test_seminar_update_submission(
+    create_ticket_mock, form_data, record_data, inspire_app
+):
     orcid = "0000-0001-5109-3700"
     user = create_user(orcid=orcid)
 
     seminar_data = {
         "control_number": 123,
         "acquisition_source": {"orcid": orcid},
-        **SEMINAR_RECORD_DATA,
+        **record_data,
     }
     create_record_factory("sem", data=seminar_data)
 
-    form_data = deepcopy({**SEMINAR_FORM_DATA, "name": "New name"})
-    form_data.pop("address")
+    update_form_data = deepcopy({**form_data, "name": "New name"})
+    expected_record_data = {**record_data, "title": {"title": "New name"}}
+
+    if "address" in update_form_data:
+        # to test deletion
+        update_form_data.pop("address")
+        expected_record_data.pop("address")
 
-    expected_record_data = {**SEMINAR_RECORD_DATA, "title": {"title": "New name"}}
-    expected_record_data.pop("address")
     with inspire_app.test_client() as client:
         login_user_via_session(client, email=user.email)
         response = client.put(
             "/submissions/seminars/123",
             content_type="application/json",
-            data=json.dumps({"data": form_data}),
+            data=json.dumps({"data": update_form_data}),
         )
     assert response.status_code == 200
 
     payload = json.loads(response.data)
     seminar_id = payload["pid_value"]
     seminar_record = SeminarsRecord.get_record_by_pid_value(seminar_id)
     seminar_record_data = {
-        key: value
-        for (key, value) in seminar_record.items()
-        if key in SEMINAR_RECORD_DATA
+        key: value for (key, value) in seminar_record.items() if key in record_data
```

Comment:
why is this needed? what extra keys does the seminar_record have?

Reply:
this is for partial assertion, keys like `control_number`

Reply:
I think it would be better to remove the fields you are sure you want to remove from the seminar_record_data or add field to the expected_record_data. For example. for the case with ```REQUIRED_SEMINAR_RECORD_DATA```, you are only asserting those fields, it would be good to assert the entire record in case there are some weird bugs and some fields that shouldn't be there are there or some that should have been removed are still there.

Reply:
yeah it would be and we have a plan to either write a helper for this or use that is already built-in to `pytest`. it wasn't straightforward that's why initially I went with this after discussing with @drjova 

anyhow this is a bug fix.